File: sdk/csharp/src/Hjtpx.Captcha.Sdk/Exceptions/CaptchaException.cs
namespace Hjtpx.Captcha.Exceptions;

public class CaptchaException : Exception
{
    public int? ErrorCode { get; }

    public CaptchaException(string message) : base(message)
    {
    }

    public CaptchaException(string message, Exception innerException) : base(message, innerException)
    {
    }

    public CaptchaException(string message, int errorCode) : base(message)
    {
        ErrorCode = errorCode;
    }

    public CaptchaException(string message, int errorCode, Exception innerException) : base(message, innerException)
    {
        ErrorCode = errorCode;
    }
}

File: sdk/php/src/Exceptions/CaptchaException.php
<?php

declare(strict_types=1);

namespace Hjtpx\Captcha\Exceptions;

use Exception;

class CaptchaException extends Exception
{
    public function __construct(string $message = '', int $code = 0, ?\Throwable $previous = null)
    {
        parent::__construct($message, $code, $previous);
    }
}
